src/transformers/processing_utils.py

```diff
@@ -790,11 +793,17 @@ class MyProcessingKwargs(ProcessingKwargs, CommonKwargs, TextKwargs, ImagesKwarg
         # get defaults from set model processor kwargs if they exist
         for modality in default_kwargs:
             default_kwargs[modality] = ModelProcessorKwargs._defaults.get(modality, {}).copy()
-            # update defaults with arguments from tokenizer init
+        # update defaults with arguments from feature processor init
+        for modality, init_kwargs in [
+            ("text", tokenizer_init_kwargs),
+            ("images", image_processor_init_kwargs),
+        ]:
+            if init_kwargs is None:
+                continue
             for modality_key in ModelProcessorKwargs.__annotations__[modality].__annotations__.keys():
                 # init with tokenizer init kwargs if necessary
-                if modality_key in tokenizer_init_kwargs:
-                    default_kwargs[modality][modality_key] = tokenizer_init_kwargs[modality_key]
+                if modality_key in init_kwargs:
+                    default_kwargs[modality][modality_key] = init_kwargs[modality_key]
```

Comment:
POI: I added handling of init kwargs for image processors cuz the layoutlm processors also check the value of `self.image_processor.apply_ocr`. None of the tests failed, but this part here needs a second look

Reply:
tests added!

will resolve this now